static-instrumenter/agent-instrumenter/src/main/java/io/opentelemetry/contrib/staticinstrumenter/ClassArchive.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.staticinstrumenter;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Enumeration;
+import java.util.Map;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipException;
+import java.util.zip.ZipOutputStream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Represents an archive storing classes (JAR, WAR). */
+class ClassArchive {
+
+  interface Factory {
+    default ClassArchive createFor(JarFile source, Map<String, byte[]> instrumentedClasses) {
+      return new ClassArchive(source, instrumentedClasses);
+    }
+  }
+
+  private static final Logger logger = LoggerFactory.getLogger(ClassArchive.class);
+
+  private final JarFile source;
+  private final Map<String, byte[]> instrumentedClasses;
+
+  ClassArchive(JarFile source, Map<String, byte[]> instrumentedClasses) {
+    this.source = source;
+    this.instrumentedClasses = instrumentedClasses;
+  }
+
+  void copyAllClassesTo(ZipOutputStream outJar) throws IOException {
```

Comment:
Not sure but since the variable name is `outJar`, should this be `JarOutputStream`? Applies throughout PR

Reply:
Interesting point! I read thru std JarOutputStream implementation and it should not make any difference, as we're copying JAR entries so magic number will be in the first entry.
There is also slight inconsistency - the tool is supposed to support both JAR and WAR format (hence: class archive) BUT there is no java model for writing WARs (only JARs). I believe then in next PRs we should switch to using ZIP (file, os) directly.

Reply:
war files are "jar" files, so shouldn't be a problem to read them with JarOutputStream

Reply:
Done :)